sklearn/ensemble/_voting.py

```diff
@@ -482,6 +508,31 @@ def get_feature_names_out(self, input_features=None):
         ]
         return np.asarray(names_out, dtype=object)
 
+    def get_metadata_routing(self):
+        """Get metadata routing of this object.
+
+        Please check :ref:`User Guide <metadata_routing>` on how the routing
+        mechanism works.
+
+        .. versionadded:: 1.4
+
+        Returns
+        -------
+        routing : MetadataRouter
+            A :class:`~sklearn.utils.metadata_routing.MetadataRouter` encapsulating
+            routing information.
+        """
+        router = MetadataRouter(owner=self.__class__.__name__)
+        router.add_self_request(self)
+
+        # `self.estimators` is a list
+        for estimator in self.estimators:
+            router.add(
+                estimator=estimator,
```

Comment:
when you add routing, the name of the argument here matters, and here you're adding all of them with the same name, which means only the last one is taken into account and the older ones will be overridden. So you need to assign them some unique name, like with their index. Like:

```suggestion
        for i, estimator in enumerate(self.estimators):
            router.add(
                **{f"estimator_{i}"=estimator},
```

And you'd use them the same way when trying to access their corresponding metadata